metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -745,20 +739,21 @@ def _select_source_nodes_with_measures(
                 nodes.append(source_node)
         return nodes
 
-    def _select_read_nodes_with_linkable_specs(
-        self, linkable_specs: LinkableSpecSet, read_nodes: Sequence[ReadSqlSourceNode]
-    ) -> List[ReadSqlSourceNode]:
+    def _select_source_nodes_with_linkable_specs(
+        self, linkable_specs: LinkableSpecSet, source_nodes: Sequence[BaseOutput]
+    ) -> Sequence[BaseOutput]:
         """Find source nodes with requested linkable specs and no measures."""
-        selected_nodes: Set[ReadSqlSourceNode] = set()
+        # Use a dictionary to deduoe for consistent ordering.
```

Comment:
nit: `dedupe` spelling

Reply:
Also curious about the use of a dict here instead of a set. Seems odd to use a dict where the values are always `None`. Set would get you the deduping, but maybe there is some ordering behavior I'm not aware of that differs between set & dict?

Reply:
Sets are unordered, so if order matters we can't use them as-is.

Do we care about duplicates at all? If so maybe we count them and log an error message if there are duplicates.

Reply:
Googling answered my question - I was thinking sets were equally as ordered as dicts in python3 but that is not the case. TIL! https://stackoverflow.com/questions/1653970/does-python-have-an-ordered-set

Reply:
Duplicates are possible as a dimension with the same name could be defined in multiple semantic models.